src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,35 @@
+import { useState, useEffect } from "react";
+import { useLocation } from 'react-router-dom'
+import * as api from '../api'
+
+export default function SingleArticle () {
+    const [article, selectArticle] = useState([])
+    const [isLoading, setIsLoading] = useState(true)
+    const location = useLocation()
+    const { id } = location.state
+
+    useEffect(() => {
+        setIsLoading(true)
+        api.fetchSingleArticle(id).then(({data}) => {
+            selectArticle(data.article.article[0])
+            setIsLoading(false)
+        })
+    }, [id]
+    )
+
+    if (isLoading) {
+        return <p className="loading">Loading...</p>
+    }
+
+    return (
+        <div className="article">
+            <h3 className="article-header">{article.title}</h3>
+            <p className="article-body">{article.body}</p>
+            <p className="article-details">author: {article.author}</p>
+            <p className="article-details">topic: {article.topic}</p>
+            <p className="article-details">comments: {article.comments_count}</p>
+            <p className="article-details">votes: {article.votes}</p>
+            <p className="article-details">created: {String(article.created_at).slice(0, 10)}</p>
```

Comment:
nicely done, there's a few packages that can do this for you from a timestamp such as moment.js if you're interested